portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -65,34 +65,23 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
     // Comment objects added to comments ArrayList will be displayed
     ArrayList<Comment> comments = new ArrayList<Comment>();
-    // count variable is the total number of comments linked with the current img
-    int count = 0;
-    // Iterating over all comments with various img tags
     for (Entity entity : results.asIterable()) {
-      String img = (String) entity.getProperty("img");
-
-      // Only add the comment if it's linked with the current img
-      if (img.equals(imgRequest)) {
-        long id = entity.getKey().getId();  
-        String name = (String) entity.getProperty("name");
-        String comment = (String) entity.getProperty("comment");
-        long timestamp = (long) entity.getProperty("timestamp");
+      long id = entity.getKey().getId();  
+      String name = (String) entity.getProperty("name");
+      String comment = (String) entity.getProperty("comment");
+      long timestamp = (long) entity.getProperty("timestamp");
         
-        Comment newComment = new Comment(id, name, comment, timestamp, img);
-        count++;
-        /* Don't add to comments to display list if the number of comments linked with the current img (count) 
-        is > the max number of comments the user wants to show (max). Keep iterating through the rest of the results, 
-        in case there are more comments linked with the current img tag, to correctly count the total (count).
-        */
-        if (count <= max) {
-            comments.add(newComment);
-        }
+      Comment newComment = new Comment(id, name, comment, timestamp, img);
+      comments.add(newComment);
+      
+      if (comments.size() >= max) {
```

Comment:
Wasn't able to catch this in the first PR

If max was 0, your comment size would be 1 since you check for comment size after you add the comment.

Reply:
Good point. We can do the check at the beginning of the for loop.

Reply:
Oooh ok got it.